src/es6/EntityFactory.js

```diff
@@ -48,24 +49,30 @@ class EntityListener {
 
 		window.D2L.Siren.EntityStore.addListener(this._href, this._token, this._onChange).then((removeListener) => {
 			this._removeListener = removeListener;
-			window.D2L.Siren.EntityStore.fetch(href, token);
+			if (entity) {
+				window.D2L.Siren.EntityStore.update(href, token, entity);
```

Comment:
This is the magic of it. If we already have the entity update it via this method instead of fetching it again.

Reply:
D2LFetch kinda already does this though right?

The simple cache middleware will likely have that entity already cached, and in the promise chain in fetch an update is run.  Does this actually change behavior?

Reply:
This is for embedded entities which the sequence API has so basically I could re-fetch them or just update them via this command.

Reply:
Right, i forgot about our embedded entities.  It does make sense to update them, I wonder if we should deep equals check to ensure we aren't notifying unecessarily

Reply:
I am taking the assumption that the entity store does the deep check. I'll check.

Reply:
It doesn't seem to :( https://github.com/Brightspace/polymer-siren-behaviors/blob/master/store/entity-store.js#L203

Reply:
Thinking about it I think polymer is smart enough to not do any updates. It will go through some of the code though. 

This can be part of another PR.

Reply:
Definitely can be another pr.  Could be backlogged in the trello board you setup.

We shouldn't think of this as being only used as polymer though.  Ideally this could be used by react, vue, lit, vanilla js, or even node.  Anything we do should adhere to generic javascript principles, not just the implementation of polymer.

Reply:
I agree!